File: solutions/lua/series/1/series.lua
return function(s, length)
  assert(s ~= "", "series cannot be empty")
  assert(length <= s:len(), "slice length cannot be greater than series length")
  assert(length ~= 0, "slice length cannot be zero")
  assert(length > 0, "slice length cannot be negative")
  local i = 0
  return function()
    i = i + 1
    if i <= s:len() - length + 1 then return s:sub(i, i + length - 1) end
  end
end

File: solutions/clojure/series/1/src/series.clj
(ns series
  (:require [clojure.string :as str]))

(defn slices [s n]
  (cond
    (empty? s) (throw (IllegalArgumentException. "series cannot be empty"))
    (zero? n) (throw (IllegalArgumentException. "slice length cannot be zero"))
    (neg? n) (throw (IllegalArgumentException. "slice length cannot be negative"))
    (> n (count s)) (throw (IllegalArgumentException. "slice length cannot be greater than series length"))

    :else (map str/join (partition n 1 s))))
